src/main/java/net/neoforged/neoforge/capabilities/IBlockCapabilityProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.capabilities;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import org.jetbrains.annotations.Nullable;
+
+@FunctionalInterface
+public interface IBlockCapabilityProvider<T, C> {
+    /**
+     * Returns the capability, or {@code null} if not available.
+     * <p>
+     * <b>If a previously returned capability is not valid anymore, or if a new capability is available,
+     * {@link Level#invalidateCapabilities(BlockPos)} MUST be called to notify the caches (see below).</b>
+     * <p>
+     * Capabilities are automatically invalidated by Forge in the following cases:
+     * <ul>
+     * <li>Chunk loads and unloads.</li>
+     * <li>Block entity loads and unloads.</li>
+     * <li>Block entity placement and destruction.</li>
+     * </ul>
+     * In ALL other cases, it is the <b>responsibility of the modder</b> to call {@link Level#invalidateCapabilities(BlockPos)},
+     * for example:
```

Comment:
Change the "for example" part into its own sentence (comma to period, capitalize "for"). You could also change it to "such as", or "such as, for example".